tests/scripts/ci_perf.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python3
+
+# This script is a simulation of what perf.groovy does in CI. It is not currently used for any official purpose, but I wanted
+# to check it in hopes it might be useful to others.
+
+import ctypes
+import json
+import argparse
+import os
+import shutil
+import subprocess
+import sys
+import glob
+from sys import version_info
+
+##########################################################################
+# Argument Parser
+##########################################################################
+
+description = 'Tool to simulate CI perf jobs'
+
+parser = argparse.ArgumentParser(description=description)
+
+#job arguments
+parser.add_argument('-job', dest='job', default=None, required=True, choices=['CoreCLR-Scenarios'])
+parser.add_argument('-isPR', dest='isPR', default=True, choices=[True, False])
+parser.add_argument('-arch', dest='arch', default='x64', choices=['x64', 'x86'])
+parser.add_argument('-os', dest='operatingSystem', default='Windows_NT', choices=['Windows_NT', 'Ubuntu16.04', 'Ubuntu14.04', 'OSX'])
+parser.add_argument('-optLevel', dest='optLevel', default='full_opt', choices=['full_opt', 'min_opt', 'tiered'])
+parser.add_argument('-jitName', dest='jitName', default='ryujit', choices=['ryujit'])
+parser.add_argument('-benchviewCommitName', dest='benchviewCommitName', default='FAKE_BV_COMMIT_NAME')
+parser.add_argument('-bvUploadSasToken', dest='bvUploadSasToken', default='FAKE_BV_SAS_UPLOAD_TOKEN')
+parser.add_argument('-uploadToBenchview', dest='uploadToBenchview', default=False, choices=[True, False])
+
+#ambient jenkins properties
+parser.add_argument('-workspace', dest='workspace', default=os.path.abspath(os.getcwd()))
+parser.add_argument('-gitBranchWithoutOrigin', dest='gitBranchWithoutOrigin', default='FAKE_BRANCHNAME')
+parser.add_argument('-gitBranch', dest='gitBranch', default='origin/FAKE_BRANCHNAME')
+parser.add_argument('-gitCommit', dest='gitCommit', default='FAKE_GIT_COMMIT')
+
```

Comment:
For anything git related, I would get it from the repo by using git commands instead of setting defaults with fake names. You can take a look at the build.py script from BenchView.

Reply:
I think there will be many cases where the state of the current branch in my enlistment isn't an accurate reproduction of what the state would be when a PR or rolling build occurs on the CI server. I'd rather make it really obvious it is a sentinel value.